### ISSUE TITLE
Error on MySQL 8

### ISSUE THREAD

**Issue by miya0001** [wp-cli/wp-cli-tests#4]
We will have an error on behat with MySQL client 8.x like following.

```
$ ./vendor/bin/behat features/post.feature 

OS:	Darwin 17.7.0 Darwin Kernel Version 17.7.0: Thu Jun 21 22:53:14 PDT 2018; root:xnu-4570.71.2~1/RELEASE_X86_64 x86_64
Shell:	
PHP binary:	/usr/local/bin/php
PHP version:	7.2.8
php.ini used:	/usr/local/etc/php/7.2/php.ini
WP-CLI root dir:	/Users/miya/wp-cli/entity-command/vendor/wp-cli/wp-cli
WP-CLI vendor dir:	/Users/miya/wp-cli/entity-command/vendor
WP_CLI phar path:	
WP-CLI packages dir:	
WP-CLI global config:	/var/folders/js/nwh8hv756_v0qqlmmgdc531h0000gn/T/wp-cli-package-test/config.yml
WP-CLI project config:	/Users/miya/wp-cli/entity-command/wp-cli.yml
WP-CLI version:	1.5.1

WordPress 4.9.7

Feature: Manage WordPress posts

  Background:          # features/post.feature:3
mysqldump: Couldn't execute 'SELECT COLUMN_NAME,                       JSON_EXTRACT(HISTOGRAM, '$."number-of-buckets-specified"')                FROM information_schema.COLUMN_STATISTICS                WHERE SCHEMA_NAME = 'wp_cli_test' AND TABLE_NAME = 'wp_commentmeta';': Unknown table 'column_statistics' in information_schema (1109)
```

Related: https://github.com/wp-cli/db-command/pull/105

**Comment by dotancohen** [comment]
I can confirm this issue on:

  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro12,1
  Processor Name:	Intel Core i5
  Processor Speed:	2.7 GHz
  Number of Processors:	1
  Total Number of Cores:	2
  L2 Cache (per Core):	256 KB
  L3 Cache:	3 MB
  Memory:	8 GB
  Boot ROM Version:	MBP121.0177.B00
  SMC Version (system):	2.28f7

**Comment by swissspidy** [comment]
Have you tried running `mysql_upgrade`? This should make sure the database is properly upgraded and that the needed tables are created.

**Comment by swissspidy** [comment]
There also seem to be some incompatibilities with MySQL 8 and PHP 7.2.8. A possible workaround for now — if `mysql_upgrade` didn't help — is adding this to your MySQL config:

```
[mysqld]
default-authentication-plugin=mysql_native_password
```

**Comment by wojsmol** [comment]
@swissspidy Other option is something like this

```SQL
ALTER USER 'username'@'localhost' IDENTIFIED WITH mysql_native_password BY 'password';
```

**Comment by dotancohen** [comment]
The problem is that MySQL client 8 cannot dump from MySQL server 5.x. The bug is filed here:
https://bugs.mysql.com/bug.php?id=89825

The workaround is to disable column statistics, which relies on 8.x:
```
$ wp db export --column-statistics=0
```

**Comment by jacklowrie** [comment]
I'm working on this in #71

**Comment by schlessera** [comment]
Closed via #71
